### ISSUE TITLE
Error on step 18

### ISSUE THREAD

**Issue by leohank** [jtamames/SqueezeMeta#22]
Hello, all
I am having an error on step 18 
[19 hours, 42 minutes, 13 seconds]: STEP18 -> CHECKING BINS: 18.checkM_batch.pl
Creating /home/biofilm/hank/SqueezeMeta/biofilm1/temp/checkm_batch
Reading /home/biofilm/sw/SqueezeMeta/scripts/../data/alltaxlist.txt
Looking for DASTool bins in /home/biofilm/hank/SqueezeMeta/biofilm1/results/DAS/biofilm1_DASTool_bins
39 bins found

Bin 1/39: maxbin.008.fasta.contigs.fa.tax
 Using profile for domain rank : Bacteria
Error running command:    export PATH="/home/biofilm/sw/SqueezeMeta/scripts/../bin/pplacer":$PATH; /home/biofilm/sw/SqueezeMeta/scripts/../bin/checkm taxon_set domain Bacteria /home/biofilm/hank/SqueezeMeta/biofilm1/data/checkm_markers/Bacteria.ms > /dev/null 2>&1 at /home/biofilm/sw/SqueezeMeta/scripts/../scripts/18.checkM_batch.pl line 136.
Stopping in STEP18 -> 18.checkM_batch.pl

How can I fix the error?

Thanks 

Hank

**Comment by fpusan** [comment]
Hi!

Here are some questions for you?

1. What is the output of the following command?

`export PATH="/home/biofilm/sw/SqueezeMeta/scripts/../bin/pplacer":$PATH; /home/biofilm/sw/SqueezeMeta/scripts/../bin/checkm taxon_set domain Bacteria /home/biofilm/hank/SqueezeMeta/biofilm1/data/checkm_markers/Bacteria.ms`

2. What OS and version of SqueezeMeta are you using?

3. Does the error also occur when running SqueezeMeta on the test data?

Best,

Fernando

**Comment by fpusan** [comment]
Closed. CheckM was not able to find the atlas Linear Algebra library in the user's computer (CentOS 7)
We were unable to reproduce this bug in our own CentOS 7 VM, so it's likely that most people won't experience it either.

In any case, the user solved it by explicitly adding the /usr/lib64/atlas/ directory to the LD_LIBRARY_PATH environment variable

` export LD_LIBRARY_PATH=/usr/lib64/atlas/:${LD_LIBRARY_PATH}`

**Comment by wangjiawen2013** [comment]
I met this error too.
here is the command:
export PATH="/home/wangjw/bin/SqueezeMeta/scripts/../bin/pplacer":$PATH; /home/wangjw/bin/SqueezeMeta/scripts/../bin/checkm taxon_set order Aeromonadales /home/wangjw/data/work/squeezeM/test/data/checkm_markers/Aeromonadales.ms

and the output is:
Traceback (most recent call last):
  File "/home/wangjw/bin/SqueezeMeta/scripts/../bin/checkm", line 45, in <module>
    from checkm import main
  File "/home/wangjw/bin/SqueezeMeta/bin/../lib/checkm/main.py", line 32, in <module>
    from checkm.treeParser import TreeParser
  File "/home/wangjw/bin/SqueezeMeta/bin/../lib/checkm/treeParser.py", line 27, in <module>
    import dendropy
ImportError: No module named dendropy

By the way, I am sure that dendropy and atlas have been installed and adding the /usr/lib64/atlas/ directory to the LD_LIBRARY_PATH environment variable:
export LD_LIBRARY_PATH=/usr/lib64/atlas/:${LD_LIBRARY_PATH}

**Comment by fpusan** [comment]
Yes, the problem is likely the python environment.
Checkm comes bundled with SqueezeMeta so no need to install it.

In commit 40b60f739b6a0c4a179afef5b16ff5da39fa6ee9 we made all the script from SqueezeMeta use the interpreters in the user's environment (e.g. /usr/bin/env python3) rather than system's default (/usr/bin/python3). We also made checkm work in python3 rather than python2. So right now you only need python3 (with the proper libraries) to run the full SqueezeMeta suite.

So long story short the latest version in master should fix your problem. Let me know if it does!

**Comment by emm-chavarria** [comment]
Hi there,
I am also facing the same error in a fresh installation running the data test provided by the developers.

Here is the command:
`../../SqueezeMeta-1.0.0/scripts/restart.pl Hadza  -step 18`

Output:
```
[0 seconds]: STEP18 -> CHECKING BINS: 18.checkM_batch.pl
Reading /home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/scripts/../data/alltaxlist.txt
Looking for DASTool bins in /home/emmanuel/Documents/Programs/databases/test/Hadza/results/DAS/Hadza_DASTool_bins
22 bins found

Bin 1/22: maxbin.006.fasta.contigs.fa.tax
 Using profile for genus rank : Alistipes
Error running command:    export PATH="/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/scripts/../bin/pplacer":$PATH; /home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/scripts/../bin/checkm taxon_set genus Alistipes /home/emmanuel/Documents/Programs/databases/test/Hadza/data/checkm_markers/Alistipes.ms > /dev/null 2>&1 at /home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/scripts/../scripts/18.checkM_batch.pl line 136.
Stopping in STEP18 -> 18.checkM_batch.pl
```

Here are my details regarding @fpusan questions:
a) Output of: `checkm taxon_set genus Alistipes /home/emmanuel/Documents/Programs/databases/test/Hadza/data/checkm_markers/Alistipes.ms`

```
Traceback (most recent call last):
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/checkm", line 45, in <module>
    from checkm import main
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/main.py", line 49, in <module>
    from checkm.plot.gcPlots import GcPlots
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/plot/gcPlots.py", line 24, in <module>
    from AbstractPlot import AbstractPlot
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/plot/AbstractPlot.py", line 22, in <module>
    from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
ImportError: No module named matplotlib.backends.backend_agg
```

b) Virtual machine with Ubuntu 19.10; SqueezeMeta 1.0.0

c) Yes, error occurs while running test data

More details.
Initially I was getting:
```
Traceback (most recent call last):
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/checkm", line 45, in <module>
    from checkm import main
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/main.py", line 23, in <module>
    import numpy as np
ImportError: No module named numpy
```
So I manually installed `numpy`, after that, I received same error now complaining about `pysam` and then `dendropy`. After installing those python libraries if I run:
`checkm taxon_set domain Bacteria /home/emmanuel/Documents/Programs/databases/test/Hadza/data/checkm_markers/Bacteria.ms`

I see this this error:
```
Traceback (most recent call last):
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/checkm", line 45, in <module>
    from checkm import main
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/main.py", line 49, in <module>
    from checkm.plot.gcPlots import GcPlots
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/plot/gcPlots.py", line 24, in <module>
    from AbstractPlot import AbstractPlot
  File "/home/emmanuel/Documents/Programs/SqueezeMeta-1.0.0/bin/../lib/checkm/plot/AbstractPlot.py", line 22, in <module>
    from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
ImportError: No module named matplotlib.backends.backend_agg
```

I also tried `alias python=python3` thinking that the version of python I'm using is not the intended. It didn't work.

Any suggestions?
Thanks

**Comment by fpusan** [comment]
Hi,
That's a detailed error report. Thanks!
What do you get when typing `which python`? The fact that all the python libraries were missing at first makes me think that you have more than one python version coexisting in your machine (perhaps bc of conda?).
Note that if you are using release 1.0.0, it should still require python2 instead of python3 for checkm.

**Comment by emm-chavarria** [comment]
Yes @fpusan I have python 3.7 and python 2.7, no conda environment.
For `which python` I get `/usr/bin/python`

SOLVED: I went back to the  INSTALL-UBUNTU14+ instructions.
It says there: 

```
### Python modules.
sudo apt-get install python3-pip python3-dev python3-scipy python3-matplotlib python3-numpy 
sudo -H python3 -m pip install dendropy pysam 
```
In order to have the needed libraries for python 2.7, as you suggested, I ran:
```
### Python modules.
sudo apt-get install python-pip python-dev python-scipy python-matplotlib python-numpy 
sudo -H python -m pip install dendropy pysam 
```

!Listo, funciona!

**Comment by OdongoIsaya** [comment]
squeezemeta stopping at step 18 seems not able to find .checKM file
